pkg/controller/certificates/authority/policies.go

```diff
@@ -31,40 +32,79 @@ import (
 type SigningPolicy interface {
 	// not-exporting apply forces signing policy implementations to be internal
 	// to this package.
-	apply(template *x509.Certificate) error
+	apply(template *x509.Certificate, usages []capi.KeyUsage, signerNotAfter time.Time, durationHint *metav1.Duration) error
 }
 
 // PermissiveSigningPolicy is the signing policy historically used by the local
 // signer.
 //
 //  * It forwards all SANs from the original signing request.
-//  * It sets allowed usages as configured in the policy.
-//  * It sets NotAfter based on the TTL configured in the policy.
+//  * It sets allowed usages as configured in the policy and based on the requested usages.
+//  * It sets NotBefore based on the Backdate configured in the policy.
+//  * It sets NotAfter based on the TTL and Backdate configured in the policy.
+//    Short lived certificates ignore Backdate and only use TTL.
+//    NotAfter is truncated to the expiration date of the signer.
 //  * It zeros all extensions.
 //  * It sets BasicConstraints to true.
 //  * It sets IsCA to false.
+//  * It validates that the signer has not expired.
 type PermissiveSigningPolicy struct {
-	// TTL is the certificate TTL. It's used to calculate the NotAfter value of
-	// the certificate.
+	// TTL is the certificate TTL. Now and TTL and Backdate are used to calculate the
+	// NotAfter value of the certificate.  Backdate is ignored when TTL is less than Short.
 	TTL time.Duration
-	// Usages are the allowed usages of a certificate.
-	Usages []capi.KeyUsage
```

Comment:
Why are we moving this?

Reply:
The structure and comment did not make sense together because we simply pass in the usages from the CSR.  So it is not allowed usages but "validated usage strings" or something like that.

Reply:
https://github.com/kubernetes/kubernetes/blob/7705b300e2085c3864bb1e49a7302bf17f080219/staging/src/k8s.io/api/certificates/v1beta1/generated.proto#L104-L107

Allowed usages are a property of a certificate. The thing that validates allowed usages is a verifier of the certificate. I admit that the godoc is not super helpful, but it seems accurate.